acceptance/tests/language/exported_resources.rb

```diff
@@ -0,0 +1,173 @@
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/facts':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/node':;
+  '#{moduledir}/yaml_terminus/lib/puppet/indirector/resource':;
+}
+file { '#{moduledir}/yaml_terminus/lib/puppet/indirector/catalog/yaml_storeconfigs.rb':
+  ensure => file,
+  content => '
+    require "puppet/indirector/catalog/yaml"
+    class Puppet::Resource::Catalog::YamlStoreconfigs < Puppet::Resource::Catalog::Yaml
+      def save(request)
+        raise ArgumentError.new("You can only save objects that respond to :name") unless request.instance.respond_to?(:name)
+        file = path(request.key)
+        basedir = File.dirname(file)
+        # This is quite likely a bad idea, since we are not managing ownership or modes.
+        Dir.mkdir(basedir) unless Puppet::FileSystem.exist?(basedir)
+        begin
+          #Puppet::Util::Yaml.dump(request.instance, file)
+          # yeah, this is the yaml terminus, but we cannot dump anonymous modules
+          #   and we have some in the catalog (request object)
+          File.open(file, "w") { |f| f.write request.instance.to_pson }
+        rescue TypeError => detail
+          Puppet.err "Could not save \#{self.name} \#{request.key}: \#{detail}"
+        end
+      end
+      def find(request)
+        nil
+      end
+    end
+  ',
+}
+file { '#{moduledir}/yaml_terminus/lib/puppet/indirector/facts/yaml_storeconfigs.rb':
+  ensure => file,
+  content => '
+    require "puppet/indirector/facts/yaml"
+    class Puppet::Node::Facts::YamlStoreconfigs < Puppet::Node::Facts::Yaml
+      def find(request)
+        nil
+      end
+    end
+  ',
+}
+file { '#{moduledir}/yaml_terminus/lib/puppet/indirector/node/yaml_storeconfigs.rb':
+  ensure => file,
+  content => '
+    require "puppet/indirector/node/yaml"
+    class Puppet::Node::YamlStoreconfigs < Puppet::Node::Yaml
+      def find(request)
+        nil
+      end
+    end
+  ',
+}
+file { '#{moduledir}/yaml_terminus/lib/puppet/indirector/resource/yaml_storeconfigs.rb':
+  ensure => file,
+  content => '
+    require "puppet/indirector/yaml"
+    require "puppet/resource/catalog"
+    class Puppet::Resource::YamlStoreconfigs < Puppet::Indirector::Yaml
+      desc "Read resource instances from cached catalogs"
+      def search(request)
+        catalog_dir = File.join(Puppet.run_mode.master? ? Puppet[:yamldir] : Puppet[:clientyamldir], "catalog", "*")
+        puts "storeconfigs filter: \#{request.options[:filter].inspect}"
+        results = Dir.glob(catalog_dir).collect { |file|
+           # yes, i know this is the yaml terminus, but we had to serialize out the catalogs in pson
+           #   see the catalog yaml_storeconfigs temrinus for why (if any)
+          catalog = Puppet::Resource::Catalog.convert_from(:pson, File.read(file))
+          if catalog.name == request.options[:host]
+            next
+          end
+          catalog.resources.select { |resource|
+            puts "resource: \#{resource.inspect}"
```

Comment:
it was for debugging, but it can't hurt?   i could make it more verbose about whom is logging (and/or use our logger).